### DISCUSSION TITLE
Checkbox and Radiobutton question

### DISCUSSION THREAD

**Discussion by ViacheslavK9** [olikraus/u8g2#2335] [Q&A]
Hi again Oliver, this is the simple one.

Could you please tell me which type of the variable we must use for the checkboxes element?

MUIF_VARIABLE("CB",&variable,mui_u8g2_u8_chkbox_wm_pi)

variable - ?

Thanks!

**Comment by ViacheslavK9** [comment]
i was looking for the kind of the array of Booleans, but realized its not the one MUIF_VARIABLE (like for the radio button), its simply many variables each with one boolean variable. Tx

**Reply by olikraus** [reply]
The checkbox expects u8 = uint8_t

**Reply by ViacheslavK9** [reply]
If i have three checkboxes on a form, do i need three variables for them as well?
For now i use three booleans and seems to be good.

**Reply by olikraus** [reply]
Yes

**Comment by ViacheslavK9** [comment]
> Yes
But why you said it expects uint8_t?

**Reply by olikraus** [reply]
Hmm i missed to mention this here: https://github.com/olikraus/u8g2/wiki/muiref#checkbox-and-radiobutton

Actually it is also encoded in the function name: `variable,mui_u8g2_u8_radio_wm_pi`. In the name the 'u8' means pointer to a uint8_t variable.

**Comment by ViacheslavK9** [comment]
> Hmm i missed to mention this here: https://github.com/olikraus/u8g2/wiki/muiref#checkbox-and-radiobutton
> 
> Actually it is also encoded in the function name: `variable,mui_u8g2_u8_radio_wm_pi`. In the name the 'u8' means pointer to a uint8_t variable.

sorry for so many questions, but why use uint8_t if we can use boolean?

**Reply by olikraus** [reply]
There is no Boolean in C

**Reply by ViacheslavK9** [reply]
bool?

**Reply by olikraus** [reply]
true, but it is a recent addition to the C standard. Maybe I can once duplicate the existing u8 functions and introduce bool instead.

**Reply by ViacheslavK9** [reply]
i see, 
still working with both bool and uint8_t though, looks good for me.
